src/Neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -33,6 +33,30 @@ public sealed partial class CryptoLib : NativeContract
 
         internal CryptoLib() : base() { }
 
+        /// <summary>
+        /// Recovers the public key from a secp256k1 signature in a single byte array format.
+        /// </summary>
+        /// <param name="messageHash">The hash of the message that was signed.</param>
+        /// <param name="signature">The 65-byte signature in format: r[32] + s[32] + v[1]. 64-bytes for eip-2098, where v must be 27 or 28.</param>
+        /// <returns>The recovered public key in compressed format, or null if recovery fails.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 10, Name = "recoverSecp256K1")]
```

Comment:
```suggestion
        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 15, Name = "recoverSecp256K1")]
```

@roman-khimov  Same price as `Verify`?

Reply:
Yeah, I expect the overhead to be about the same, certainly not `1 << 10`. Can be retested to ensure it's comparable.

Reply:
@superboyiii